src/DependencyInjection/JaegerExtension.php

```diff
@@ -18,13 +18,40 @@ public function load(array $configs, ContainerBuilder $container)
             new FileLocator(__DIR__ . '/../Resources')
         );
         $loader->load('services.yml');
-
         $configuration = new Configuration();
         $config = $this->processConfiguration($configuration, $configs);
-
         if ($this->isConfigEnabled($container, $config['denylist'])) {
             $container->setParameter('jaeger.sampler.operation_denylist', $config['denylist']['operation_names']);
             $loader->load('denylist.yml');
         }
+        if ($this->isConfigEnabled($container, $config['name_generator'])) {
+            $container->setParameter('jaeger.name.max_length', (int)$config['name_generator']['max_length']);
+            foreach ($config['name_generator']['request'] as $item => $alias) {
+                if ($container->has(\sprintf('jaeger.name.generator.%s', $alias))) {
+                    $container->getDefinition('jaeger.name.generator.request')->addMethodCall(
+                        'add',
+                        [$item, new Reference(\sprintf('jaeger.name.generator.%s', $alias))]
+                    );
+                } else {
+                    $container->getDefinition('jaeger.name.generator.request')->addMethodCall(
+                        'add',
+                        [$item, new Reference($alias)]
+                    );
+                }
+            }
+            foreach ($config['name_generator']['command'] as $item => $alias) {
+                if ($container->has(\sprintf('jaeger.name.generator.%s', $alias))) {
```

Comment:
I think this would be better to read (same for request generators):

```php
        foreach ($config['name_generator']['command'] as $regexp => $customGeneratorId) {
            $shortenedGeneratorId = \sprintf('jaeger.name.generator.%s', $customGeneratorId);
            if ($container->has($shortenedGeneratorId)) {
                $customGeneratorId = $shortenedGeneratorId;
            }
            
            $container->getDefinition('jaeger.name.generator.command')
                ->addMethodCall(
                    'add',
                    [$regexp, new Reference($customGeneratorId)]
                );
        }
```